src/editor.c

```diff
@@ -3526,10 +3526,13 @@ static void auto_multiline(GeanyEditor *editor, gint cur_line)
 		{ /* we are on the second line of a multi line comment, so we have to insert white space */
 			whitespace = " ";
 		}
-
-		if (style == SCE_D_COMMENTNESTED)
-			continuation = "+"; /* for nested comments in D */
-
+		else if (!(g_str_has_prefix(previous_line + i, continuation) &&
```

Comment:
I'd just suggest inverting the condition (i.e. removing the initial `!`) and adding the 3 lines after this block inside the block. This way you avoid the duplicated `g_free(previous_line);` and the extra `return`.

Reply:
Apart from that the PR looks fine to me and is useful, I also did run into this issue in the past.

Reply:
That would break the first if clause before the else if on this line.

Reply:
Ah, right, sorry, I didn't think about that.

Reply:
No problem, thanks for reviewing.